src/secure_channel.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! Agent is a module used to connect KBS (Relying Party)
+//! for confidential resources
+
+use std::collections::HashMap;
+
+use anyhow::*;
+use serde::{Deserialize, Serialize};
+use tokio::fs;
+use tonic::transport::Channel;
+
+use self::get_resource::{
+    get_resource_service_client::GetResourceServiceClient, GetResourceRequest,
+};
+
+mod get_resource {
+    #![allow(unknown_lints)]
+    #![allow(clippy::derive_partial_eq_without_eq)]
+    tonic::include_proto!("getresource");
+}
+
+/// Attestation Agent's GetResource gRPC address.
+/// It's given <https://github.com/confidential-containers/attestation-agent#run>
+pub const AA_GETRESOURCE_ADDR: &str = "http://127.0.0.1:50001";
+
+/// The resource description that will be passed to AA when get resource.
+#[derive(Serialize, Deserialize, Debug)]
+struct ResourceDescription {
+    name: String,
+    optional: HashMap<String, String>,
+}
+
+impl ResourceDescription {
+    /// Create a new ResourceDescription with resource name.
+    pub fn new(name: &str, optional: HashMap<String, String>) -> Self {
+        ResourceDescription {
+            name: name.to_string(),
+            optional,
+        }
+    }
+}
+
+/// SecureChannel to connect with KBS
+pub struct SecureChannel {
+    /// Get Resource Service client.
+    client: GetResourceServiceClient<Channel>,
+    kbc_name: String,
+    kbc_uri: String,
+}
+
+impl SecureChannel {
+    /// Create a new [`SecureChannel`], the input parameter:
+    /// * `aa_kbc_params`: s string with format `<kbc_name>::<kbs_uri>`.
+    pub async fn new(aa_kbc_params: &str) -> Result<Self> {
```

Comment:
In the overall flow where is  `aa_kbc_params` specified? Is it included in hardware measurement of the kata-cc vm?

Reply:
Hi, here is a brief call routine
`SecureChannel::new(aa_kbc_params)` in `mage-rs` <- `ImageClient::pull_image()` in `mage-rs` <- `AGENT_CONFIG` in [`kata-agent`](https://github.com/confidential-containers/kata-containers-CCv0/blob/CCv0/src/agent/src/image_rpc.rs#L249) <- [From Guest commandline](https://github.com/confidential-containers/kata-containers-CCv0/blob/CCv0/src/agent/src/main.rs#L85)

For the second question, sry for not deep dive into this now, but [this issue](https://github.com/kata-containers/kata-containers/issues/5178) may help you

Reply:
Thanks for the description, @Xynnn007. One thing that is not clear is whether which KBS kata-agent/image-rs talks to is measured.

Reply:
Hi, @anakrish , in the current deployment model, KBS is controlled by the tenant and deployed in the security environment outside the cloud by the tenant, so KBS does not need to be measured. In other words, KBS actually plays the role of "rely party and verifier".